crates/transaction-pool/src/validate/eth.rs

```diff
@@ -326,6 +340,14 @@ where
         origin: TransactionOrigin,
         transaction: Self::Transaction,
     ) -> TransactionValidationOutcome<Self::Transaction> {
+        #[cfg(feature = "optimism")]
+        if transaction.is_deposit() {
+            return TransactionValidationOutcome::Invalid(
+                transaction,
+                InvalidTransactionError::PooledDepositTx.into(),
+            )
+        }
```

Comment:
this seems fine for now, but I'd prefer a new transaction validator type instead of feature gating certain checks

I should be able to make the eth checks reusable though